src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -217,6 +224,24 @@ public PersonalAccessToken getCredentials() {
             ), withId(credentialsId));
     }
 
+    /**
+     * @param hooksRootUrl a custom root URL, to be used in hooks instead of {@link Jenkins#getRootUrl()}.
+     * Set to {@code null} for default behavior.
+     */
+    @DataBoundSetter
+    public void setHooksRootUrl(String hooksRootUrl) {
+        this.hooksRootUrl = hooksRootUrl;
+    }
+
+    /**
+     * @return the custom root URL, to be used in hooks instead of {@link Jenkins#getRootUrl()}.
+     * Can be either a root URL with its trailing slash, or {@code null}.
+     */
+    @CheckForNull
+    public String getHooksRootUrl() {
```

Comment:
```suggestion
    public String getHooksRootUrl() {
    if (hooksRootUrl == null || hooksRootUrl.isEmpty()) {
        return Jenkins.get().getRootUrl();
    }
```

Reply:
I think I prefer keeping the getter return value close enough to what's actually stored in the `GitLabServer` config object, and the *"what default value to fall back to"* logic in the `GitLabHookCreator.getHookUrl(...)` implementation.

Saying that made me think of doing 2e9b787 to simplify things a tiny bit.

Reply:
I feel the server configuration should also have a field like SCMTraits where user can add their configurations rather than a default empty field. Well, these changes might require a bit of extra remodelling so for now I think we are good.

Reply:
a default empty field is how both GitHub- and Bitbucket branch source plugin handles the custom hook url 👍
I had the pretty written the same words as @thomasgl-orange in my pending review to this comment 😆